src/CPUSolver.cpp

```diff
@@ -1963,7 +1951,7 @@ void CPUSolver::addSourceToScalarFlux() {
     for (int e=0; e < _num_groups; e++) {
       _scalar_flux(r, e) /= (sigma_t[e] * volume);
       _scalar_flux(r, e) += FOUR_PI * _reduced_sources(r, e) / sigma_t[e];
-      if (_scalar_flux(r, e) <= 0.0) {
+      if (_scalar_flux(r, e) < 0.0) {
```

Comment:
It is dangerous to compare the equality between a float variable and 0.0. There are several other places with this danger in the whole code.
It should be abs(f)<EPSILON where EPSILON is a tiny value set for different usages.

Reply:
Here we are checking for the sign only, not proximity to 0 so it should be ok

Reply:
It is just a reminder. We'd better fix it when seeing it next time.